crates/chia-datalayer/src/merkle.rs

```diff
@@ -0,0 +1,1997 @@
+#[cfg(feature = "py-bindings")]
+use pyo3::{
+    buffer::PyBuffer,
+    exceptions::{PyAttributeError, PyValueError},
+    pyclass, pymethods, PyResult, Python,
+};
+
+use clvmr::sha2::Sha256;
+use num_traits::ToBytes;
+use std::cmp::Ordering;
+use std::collections::{HashMap, HashSet, VecDeque};
+use std::iter::zip;
+use std::mem::size_of;
+use std::ops::Range;
+
+type TreeIndex = u32;
+type Parent = Option<TreeIndex>;
+type Hash = [u8; 32];
+type KvId = i64;
+
+// assumptions
+// - root is at index 0
+// - any case with no keys will have a zero length blob
+
+const fn range_by_length(start: usize, length: usize) -> Range<usize> {
```

Comment:
Is this really an improvement?

Reply:
removed in [`9ab4599` (#673)](https://github.com/Chia-Network/chia_rs/pull/673/commits/9ab45995f33984bd2f02728b1343bf3c8fac7b0a)